### DISCUSSION TITLE
Relacionamento muito para muitos

### DISCUSSION THREAD

**Discussion by CVieiraSantos** [balta-io/forum#17] [Tira Dúvidas] [answered]
No módulo 3 ao criar a classe PostMap e realizar o relacionamento muito para muitos, seria necessário fazer a mesma codificação na classe TagMap por exemplo para realizar este tipo de relacionamento e assim funcionar o mesmo?
Ou o framework já entende se fez o relacionamento na classe PostMap é como se também estivesse feito na classe TagMap?
Porque no módulo 3 eu não vi o mapeamento da classe Tag.

**Comment by andrebaltieri** [comment, marked as answer]
@CVieiraSantos bom dia, somente em uma das duas já basta :)

**Reply by CVieiraSantos** [reply]
@andrebaltieri bom dia. Então eu fiz essa pergunta porque estou criando um outro projeto por fora que eu tenho uma classe produto e outra classe categoria. Eu criei uma classe produtomap e fiz todo aquele processo que você fez no curso, e aí fiz o relacionamento muito muito para muitos na classe produtomap. Só que mesmo criando de forma automática pela migration, eu percebi que a classe categoria ela foi gerada de qualquer jeito pelo Framework, por isso que te perguntei se eu precisaria mapear também.
Daí quando eu crio um post, e dou um get, lá no final vem da seguinte forma "categoria":null
Porque isso acontece, será que é porque não foi mapeada a classe categoria?

**Reply by andrebaltieri** [reply]
Isto provavelmente ocorreu pelo fato de você usar o `new` na hora de criar uma categoria. Como eu comento no curso, sempre que você faz isto, o EF entende que é um novo objeto e deve ser persistido.

No caso, você precisa buscar a categoria no banco e depois atrelar ela ao objeto.
